File: java/src/array/PrimeListUnderNumber.java
package array;

import java.util.ArrayList;
import java.util.List;

public class PrimeListUnderNumber {
    public static void main(String[] args) {
        System.out.println(findPrimeListUnderNumber(20));
    }

    static List<Integer> findPrimeListUnderNumber(int number) {
        List<Integer> list = new ArrayList<>();

        for (int i = 2; i <= number; i++) {
            boolean isPrime = true;

            for (int j : list) {
                if (j * j > i)
                    break;
                if (i % j == 0)
                {
                    isPrime = false;
                    break;
                }

            }

            if (isPrime) {
                list.add(i);
            }
        }

        return list;
    }
}

File: python/array/prime_list_under_number.py
# 소수 나열하기

def find_prime_list_under_number(number):
    prime_list = []
    for i in range(2, number + 1):
        for j in prime_list:
            if j * j <= i and i % j == 0:
                break
        else:
            prime_list.append(i)

    return prime_list


if __name__ == "__main__":
    print(find_prime_list_under_number(20))
